vulcanai/models/utils.py

```diff
@@ -4,13 +4,75 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.autograd import Variable
+from torch.utils.data import DataLoader, TensorDataset
 
 import numpy as np
 import math
 from sklearn.metrics import confusion_matrix
 from sklearn.preprocessing import LabelBinarizer
 from collections import OrderedDict as odict
+from collections import defaultdict
 
+def get_probs(network, loader, index_to_iter, ls_feat_vals):
+    """
+    Returns probability for each object within loader based on output from training neural network
+    Parameters:
+        network : vulcan.model
+	    training vulcan network
+	loader : torch.dataloader
+	    dataloader containing validation set
+	index_to_iter : string
+	    feature to iterate through adjusting values
+	ls_feat_vals : list
+	    values to iterate through for feature in index_to_iter
+    Returns:
+	dct_scores : dictionary
+	    dictionary of scores
+    """
+    dct_scores = defaultdict()
+    for index in range(len(loader)):
+        dct_scores[index] = {}
+    for index in range(len(loader)):
+        nn_subjFeat = torch.Tensor(loader.dataset[index][0].numpy())
```

Comment:
To be honest, I don't know if it's necessary but the biggest issue I had was here so I just left it. I can test some other things out to see if its necessary.

Reply:
Looking through it again, it does make it easier to update the loader with new values. So once we start iterating through values its easy just to update `nn_subjFeat`, in my opinion. I'll test if it's necessary and if not we can decide which route we'd like to go.

Reply:
Okay, just tested it isn't needed. So we can leave it or change it to the following:

`
input_loader = DataLoader(TensorDataset(loader.dataset[index][0].unsqueeze(0), nn_subjLabel.unsqueeze(0)))
subjProb = network.forward_pass(data_loader=input_loader, transform_outputs=False)
`
nn_subjLabel will need to stay because we need to change it a LongTensor.

Then when changing the value of the given feature:

`
loader.dataset[index][0][index_to_iter] = newVal
`

Reply:
I think we should update it, in the spirit of trying to optimize our code a little. Not sure how much of a difference it would make but it might make some, especially as it's in a loop.

Reply:
Done